tools/build-tools/src/buildVersion/buildVersion.ts

```diff
@@ -90,8 +91,60 @@ export function getSimpleVersion(file_version: string, arg_build_num: string, ar
     return fullVersion;
 }
 
+type TagPrefix = string | undefined | "client" | "server";
+
+/**
+ * Extracts versions from the output of `git tag -l` in the working directory. The returned array will be sorted
+ * ascending by semver version rules.
+ *
+ * @param prefix - the tag prefix to filter the tags by (client, server, etc.).
+ * @returns an array of versions extracted from the output of `git tag -l`.
+ */
+function getVersions(prefix: TagPrefix) {
+    const raw_tags = child_process.execSync(`git tag -l`, { encoding: "utf8" });
+    const tags = raw_tags.split(/\s+/g).map(t => t.trim());
+    const filtered = tags.filter(v => v.startsWith(`${prefix}_v`));
+    return getVersionsFromStrings(prefix, filtered);
+}
+
+/**
+ * Extracts versions from an array of strings, sorts them according to semver rules, and returns the sorted array.
+ *
+ * @param prefix - the tag prefix to filter the tags by (client, server, etc.).
+ * @param tags - an array of strings.
+ * @returns an array of versions extracted from the provided tags.
+ */
+export function getVersionsFromStrings(prefix: TagPrefix, tags: string[]) {
+    let versions = tags.map((tag) => {
+        const split = tag.split(`${prefix}_v`);
```

Comment:
Since the tag should be filtered already, we can use substring instead of split?

Reply:
Fixed.